packages/ai-jsx/src/experimental/next/index.tsx

```diff
@@ -98,30 +125,34 @@ export const jsx = asJsxBoundary(function jsx(
     prefix,
     replace,
   }: {
-    prefix?: ReactModule.ReactNode[];
+    prefix?: ReactNode[];
     /**
      * Replaces the entire existing stream (rather than appending to it). Must be used
      * when the stream is complete to indicate that the stream has finished.
      */
-    replace: (value: ReactModule.ReactNode) => ReactModule.ReactNode;
-  }): Promise<ReactModule.ReactNode> {
+    replace: (value: ReactNode) => ReactNode;
+  }): Promise<ReactNode> {
     const nextResult = await asyncIterator.next();
 
     if (nextResult.done) {
-      return replace(nextResult.value.map(unwrapReact));
+      const finalResult = nextResult.value.map(unwrapReact);
+      const finalText = nextResult.value.filter((v) => typeof v === 'string').join('');
```

Comment:
What are the elements that would be filtered out here?

Reply:
Any React elements produced by AI.JSX. (e.g. `<Recipe>` and friends, or whatever else replaces that in a JIT UI world)